### ISSUE TITLE
buck2_re_client server and config?

### ISSUE THREAD

**Issue by teh** [facebookincubator/buck2#76]
Hi - I can a config section for `buck2_re_client` in the code, and traced the various components like `perform_cache_upload`  but it's not clear to me what that cache protocol looks like. Is this code usable in the open source release?

**Comment by cjhopman** [comment]
This should be mostly working, you can see some of the support for it here: https://github.com/facebookincubator/buck2/tree/main/remote_execution/oss

This is using the remote execution apis described here: https://github.com/bazelbuild/remote-apis/blob/main/build/bazel/remote/execution/v2/remote_execution.proto. AIUI, there are several open source implementations of the server components of that api (e.g. https://github.com/buildbarn/bb-remote-execution).

It doesn't look like there's an example or documentation yet about how to set that up in the open source release, but I think you can expect that that will be coming soon.

**Comment by ndmitchell** [comment]
It mostly works, but there are a few patches and configuration steps needed:

1. We are still in the process of splitting file vs directory artifacts, which is a requirement for the Bazel RE API.
2. We use SHA1, which only BuildBarn supports.
3. There is some configuration in the RE settings that needs passing which we currently hardcode to our internal API.

So we have done a small build using BuildBarn, and expect to have instructions on how to do real builds in the next month or so, but for the moment, it's probably infeasible to do so.

**Comment by teh** [comment]
I can see an example here now https://github.com/facebook/buck2/tree/main/examples/remote_execution/engflow so I think the issue can be closed.

One thing I haven't figured out is how to just use a shared action cache without a remote executor. Running without `engine_address` throws an error.

```
Jobs: In progress: 78. Finished: 42. Cache hits: 0%. Time elapsed: 2.2s
````

^ the Cache hits is always zero - I assume that's because there is no local cache?

**Comment by ndmitchell** [comment]
Yep, the shared action cache currently requires the remote executor. We are looking at whether a local shared executor might be a good idea, e.g. #105
